src/convertToObject.js

```diff
@@ -3,16 +3,34 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesArray = sourceString.split(';');
+  const stylesObject = {};
```

Comment:
This empty object is redundant, you can pass it as starting value in the reduce and immediately return reduced val:
```js
 return stylesArray.reduce(
    (styles, style) => {
      const styleParts = style.split(':');
      const key = styleParts[0].trim();

      if (key !== '') {
        styles[key] = styleParts[1].trim();
      }

      return styles;
    }
    , {});
```

Reply:
Also name of the styleParts could be more meaningful, `const propertyAndValue = style.split(':');`

Reply:
I personally wouldn't declare any variables here and just return
```js
return sourceString.split(';').reduce(
    (styles, style) => {
      const styleParts = style.split(':');
      const key = styleParts[0].trim();

      if (key !== '') {
        styles[key] = styleParts[1].trim();
      }

      return styles;
    }
    , {});
```

Reply:
And finally I would always return a new object with previous styles spread and new optional prop
```js
return {
  ...styles,
  ...(prop !== "" && { [prop]: propertyAndValue[1].trim() }),
};
```

so finally the function could look like this:
![image](https://github.com/mate-academy/js_style-to-object/assets/87986955/21a2bfbc-5850-4275-9860-e51f121d9d85)

Reply:
I shorten code.